fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/DelinquencyReadPlatformServiceImpl.java

```diff
@@ -0,0 +1,105 @@
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.apache.fineract.portfolio.delinquency.data.DelinquencyBucketData;
+import org.apache.fineract.portfolio.delinquency.data.DelinquencyRangeData;
+import org.apache.fineract.portfolio.delinquency.domain.DelinquencyBucket;
+import org.apache.fineract.portfolio.delinquency.domain.DelinquencyBucketRepository;
+import org.apache.fineract.portfolio.delinquency.domain.DelinquencyRange;
+import org.apache.fineract.portfolio.delinquency.domain.DelinquencyRangeRepository;
+import org.apache.fineract.portfolio.delinquency.exception.DelinquencyBucketNotFoundException;
+import org.apache.fineract.portfolio.delinquency.mapper.DelinquencyBucketMapper;
+import org.apache.fineract.portfolio.delinquency.mapper.DelinquencyRangeMapper;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class DelinquencyReadPlatformServiceImpl implements DelinquencyReadPlatformService {
+
+    private final JdbcTemplate jdbcTemplate;
+    private final DelinquencyRangeRepository repositoryRange;
+    private final DelinquencyBucketRepository repositoryBucket;
+    private final DelinquencyRangeMapper mapperRange;
+    private final DelinquencyBucketMapper mapperBucket;
+
+    @Override
+    public Collection<DelinquencyRangeData> retrieveAllDelinquencyRanges() {
+        final List<DelinquencyRange> delinquencyRangeList = repositoryRange.findAll();
+        return mapperRange.map(delinquencyRangeList);
+    }
+
+    @Override
+    public DelinquencyRangeData retrieveDelinquencyRange(Long delinquencyRangeId) {
+        DelinquencyRange delinquencyRangeList = repositoryRange.getReferenceById(delinquencyRangeId);
+        return mapperRange.map(delinquencyRangeList);
+    }
+
+    @Override
+    public Collection<DelinquencyBucketData> retrieveAllDelinquencyBuckets() {
+        final List<DelinquencyBucket> delinquencyRangeList = repositoryBucket.findAll();
+        return mapperBucket.map(delinquencyRangeList);
+    }
+
+    @Override
+    public DelinquencyBucketData retrieveDelinquencyBucket(Long delinquencyBucketId) {
+        try {
+            final DelinquencyBucket delinquencyBucket = repositoryBucket.getReferenceById(delinquencyBucketId);
+            final DelinquencyBucketData delinquencyBucketData = mapperBucket.map(delinquencyBucket);
+
+            final DelinquencyBucketMappingMapper rmMappings = new DelinquencyBucketMappingMapper();
+            final String sql = "select " + rmMappings.schema() + " where buc.id = ?";
+            List<DelinquencyRangeData> ranges = this.jdbcTemplate.query(sql, rmMappings, new Object[] { delinquencyBucketId }); // NOSONAR
```

Comment:
Minor: the query params can be unwrapped....

Reply:
Done!